### PR TITLE
Evm circuit push

### PR DESCRIPTION
**Pull request by miha-stopar**
PUSH opcode first version.

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/op_execution/push.rs [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+impl<F: FieldExt> PushGadget<F> {
+    fn assign_success(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        op_execution_state: &mut OpExecutionStateInstance,
+        execution_step: &ExecutionStep,
+    ) -> Result<(), Error> {
+        op_execution_state.global_counter += 1;
+        op_execution_state.program_counter +=
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit/op_execution/push.rs:247]: push1 operation increases program_counter + 2 which is different than other op code

File: zkevm-circuits/src/evm_circuit/op_execution/push.rs [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            (execution_step.opcode - 95) as usize;
+        op_execution_state.stack_pointer += 1;
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit/op_execution/push.rs:249]: should be stack_pointer -= 1

File: zkevm-circuits/src/evm_circuit/op_execution/push.rs (hunk 1 above)
Comment (han0110) [reply zkevm-circuits/src/evm_circuit/op_execution/push.rs]: Fixed. Thanks!

File: zkevm-circuits/src/evm_circuit/op_execution/push.rs (hunk 2 above)
Comment (han0110) [reply zkevm-circuits/src/evm_circuit/op_execution/push.rs]: Fixed. Thanks!

**Pull request merged**